packages/packagers/js/src/ScopeHoistingPackager.js

```diff
@@ -741,10 +741,22 @@ ${code}
         this.hoistedRequires.set(dep.id, hoisted);
       }
 
-      hoisted.set(
-        resolvedAsset.id,
-        `var $${publicId} = parcelRequire(${JSON.stringify(publicId)});`,
+      let resolvedAssetUsedSymbols = nullthrows(
+        this.bundleGraph.getUsedSymbols(resolvedAsset),
       );
+
+      if (
+        resolvedAssetUsedSymbols.has(exportSymbol) ||
+        nullthrows(this.bundleGraph.getUsedSymbols(dep)).has(exportSymbol) ||
+        (exportSymbol === 'default' &&
+          resolvedAssetUsedSymbols.has(exportSymbol)) ||
+        (exportSymbol === '*' && resolvedAssetUsedSymbols.size > 0)
```

Comment:
Maybe a comment (or temporary variable) that explains this large expression will help 🙂